Mathlib/Geometry/Convex/Cone/Basic.lean

```diff
@@ -382,6 +382,101 @@ instance instAddCommSemigroup : AddCommSemigroup (ConvexCone R M) where
   add_assoc _ _ _ := SetLike.coe_injective <| add_assoc _ _ _
   add_comm _ _ := SetLike.coe_injective <| add_comm _ _
 
+section Generating
+
+/-- A convex cone `C` is generating if its linear span is the entire `R`-module `M`.
+
+`IsGenerating` is equivalent to `IsReproducing` modulo some conditions.
+See `IsReproducing.isGenerating` and `IsGenerating.isReproducing` for details. -/
+@[simp] def IsGenerating (C : ConvexCone R M) : Prop :=
+  Submodule.span R (C : Set M) = ⊤
+
+/-- The whole `R`-module `M` (viewed as the top convex cone) is generating. -/
+theorem isGenerating_top : (⊤ : ConvexCone R M).IsGenerating := by
+  simp
+
+/-- A convex cone containing a generating cone is also a generating cone. -/
+theorem IsGenerating.mono (h : C₁ ≤ C₂) (hgen : C₁.IsGenerating) : C₂.IsGenerating := by
+  rw [IsGenerating, ← top_le_iff] at hgen ⊢
+  exact hgen.trans (Submodule.span_mono h)
+
+/-- A convex cone is reproducing if its set of element differences equals the entire module,
+i.e., every element of `M` can be written as a difference of two elements of `C`.
+
+This is an alternative notion of generating (`IsGenerating`) which is equivalent in many cases.
+Note: requires additive inverses and hence `[AddCommGroup M]` in place of `[AddCommMonoid M]`. -/
+def IsReproducing {R : Type*} {M : Type*} [Semiring R] [PartialOrder R] [AddCommGroup M]
```

Comment:
Here and below, I think we don't need to be so explicit on the doc string about the assumptions.
```suggestion
See also `ConvexCone.IsGenerating`. -/
@[simp] def IsReproducing {R M : Type*} [Semiring R] [PartialOrder R] [AddCommGroup M]
```